meta-lmp-base/recipes-crypto/cryptsetup/cryptsetup/0001-reencrypt-online-reencryption-with-TPM-PKCS11-tokens.patch

```diff
@@ -0,0 +1,276 @@
+
+Whenever a TPM or PKCS11 key have been enrolled, they will be used for
+re-encryption without requiring a passphrase.
+
+Enable similar functionality for the repair action which might be needed
+if re-rencryption is interrupted.
+
+Upstream-Status: Inappropiate [LmP use case]
+
+Signed-off-by: Jorge Ramirez-Ortiz
+---
+ lib/Makemodule.am      |  2 ++
+ lib/keyslot_context.c  |  2 +-
+ lib/keyslot_context.h  |  6 +++++
+ lib/libcryptsetup.sym  |  1 +
+ lib/utils_passphrase.c | 55 ++++++++++++++++++++++++++++++++++++++++++
+ lib/utils_passphrase.h | 30 +++++++++++++++++++++++
+ src/cryptsetup.c       |  9 ++++++-
+ src/utils_reencrypt.c  | 38 +++++++++++++++++++++++++++--
+ 8 files changed, 139 insertions(+), 4 deletions(-)
+ create mode 100644 lib/utils_passphrase.c
+ create mode 100644 lib/utils_passphrase.h
+
+diff --git a/lib/Makemodule.am b/lib/Makemodule.am
+index ae5fab99..2de963a5 100644
+--- a/lib/Makemodule.am
++++ b/lib/Makemodule.am
+@@ -75,6 +75,8 @@ libcryptsetup_la_SOURCES = \
+ 	lib/tcrypt/tcrypt.c		\
+ 	lib/keyslot_context.h		\
+ 	lib/keyslot_context.c		\
++	lib/utils_passphrase.h		\
++	lib/utils_passphrase.c		\
+ 	lib/luks1/af.h			\
+ 	lib/luks1/af.c			\
+ 	lib/luks1/keyencryption.c	\
+diff --git a/lib/keyslot_context.c b/lib/keyslot_context.c
+index cb804e8f..817f4791 100644
+--- a/lib/keyslot_context.c
++++ b/lib/keyslot_context.c
+@@ -286,7 +286,7 @@ static int get_luks2_volume_key_by_token(struct crypt_device *cd,
+ 	return get_luks2_key_by_token(cd, kc, keyslot, CRYPT_DEFAULT_SEGMENT, r_vk);
+ }
+ 
+-static int get_passphrase_by_token(struct crypt_device *cd,
++int get_passphrase_by_token(struct crypt_device *cd,
+ 	struct crypt_keyslot_context *kc,
+ 	const char **r_passphrase,
+ 	size_t *r_passphrase_size)
+diff --git a/lib/keyslot_context.h b/lib/keyslot_context.h
+index 306e0a56..27924bc1 100644
+--- a/lib/keyslot_context.h
++++ b/lib/keyslot_context.h
+@@ -150,4 +150,10 @@ void crypt_keyslot_unlock_by_vk_in_keyring_internal(struct crypt_keyslot_context
+ 
+ const char *keyslot_context_type_string(const struct crypt_keyslot_context *kc);
+ 
++/* TPM2 re-rencrypt */
++int get_passphrase_by_token(struct crypt_device *cd,
++			    struct crypt_keyslot_context *kc,
++			    const char **r_passphrase,
++			    size_t *r_passphrase_size);
++
+ #endif /* KEYSLOT_CONTEXT_H */
+diff --git a/lib/libcryptsetup.sym b/lib/libcryptsetup.sym
+index 68faffe6..33945dd2 100644
+--- a/lib/libcryptsetup.sym
++++ b/lib/libcryptsetup.sym
+@@ -178,4 +178,5 @@ CRYPTSETUP_2.7 {
+ 		crypt_resume_by_keyslot_context;
+ 		crypt_set_keyring_to_link;
+ 		crypt_wipe_hw_opal;
++		crypt_get_passphrase_from_token;
+ } CRYPTSETUP_2.6;
+diff --git a/lib/utils_passphrase.c b/lib/utils_passphrase.c
+new file mode 100644
+index 00000000..5903caeb
+--- /dev/null
++++ b/lib/utils_passphrase.c
+@@ -0,0 +1,55 @@
++/*
++ * libcryptsetup - cryptsetup library
++ *
++ * Copyright (C) 2023 Jorge Ramirez-Ortiz <jorge@foundries.io>
```

Comment:
copyright is actually to foundries, you're the author (same for other copyright lines).

Reply:
sure makes sense to me - I pretty much copy-pasted what Fathi did for resize-helper (his name and his linaro email).